File: Data Structures & Algorithms/anagram-groups/submission-0.py
from collections import defaultdict
from typing import List


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        a = defaultdict(list) # dictionary where values initialised as lists
        for word in strs:
            s = "".join(sorted(word))
            a[s].append(word)

        res = list(a.values()) # returns all values in list
        return res



strs = ["act","pots","tops","cat","stop","hat"]
solution = Solution()
print(solution.groupAnagrams(strs))

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cs
public class Solution {
    public int FindDuplicate(int[] nums) {
        Dictionary<int, int> values = new Dictionary<int, int>();

        for(int i = 0; i < nums.Length; i++){
            if(values.ContainsValue(nums[i])){
                return nums[i];
            }
            else {
                values[i] = nums[i];
            }
        }

        return 0;
    }
}
